bqskit/runtime/detached.py

```diff
@@ -0,0 +1,571 @@
+        w_id = result.completed_by
+        m_id = (w_id - self.lower_id_bound) // self.step_size
+        self.manager_idle_resources[m_id] += 1
+        dest_w_id = result.return_address.worker_id
+
+        # If it isn't for me
+        if dest_w_id != -1:
+            w_id = result.return_address.worker_id - self.lower_id_bound
+            m_id = w_id // self.step_size
+            assert m_id >= 0 and m_id < len(self.managers)
+            m = (self.managers[m_id], RuntimeMessage.RESULT, result)
+            self.outgoing.append(m)
+
+        # If it is for me
+        else:
+            mailbox_id = result.return_address.mailbox_index
+            if mailbox_id not in self.mailboxes:
+                return  # Silently discard results from cancelled tasks
+
+            box = self.mailboxes[mailbox_id]
+            box[0] = result.result
+            t_id = self.mailbox_to_task_dict[mailbox_id]
+            self.logger.info(f'Finished: {t_id}.')
+            if box[1]:
+                m = (self.tasks[t_id][1], RuntimeMessage.RESULT, box[0])
+                self.outgoing.append(m)
+                self.clients[self.tasks[t_id][1]].remove(t_id)
+                # self.tasks.pop(t_id)
+                self.mailboxes.pop(mailbox_id)
+                # self.mailbox_to_task_dict.pop(mailbox_id)
+
+    def _handle_status(self, conn: Connection, request: uuid.UUID) -> None:
+        """Inform the client if the task is finished or not."""
+        if request not in self.clients[conn] or request not in self.tasks:
+            conn.send((RuntimeMessage.STATUS, CompilationStatus.UNKNOWN))
+
+        mailbox_id = self.tasks[request][0]
+        box = self.mailboxes[mailbox_id]
+
+        if box[0] is None:
+            conn.send((RuntimeMessage.STATUS, CompilationStatus.STARTED))
+
+        else:
+            conn.send((RuntimeMessage.STATUS, CompilationStatus.DONE))
+
+    def _handle_cancel_comp_task(self, request: uuid.UUID) -> None:
+        """Cancel a compilation task in the system."""
+        self.logger.info(f'Cancelling: {request}.')
+
+        # Remove task from server data
+        mailbox_id, client_conn = self.tasks[request]  # .pop(request)
+        # self.mailbox_to_task_dict.pop(mailbox_id)
+        self.mailboxes.pop(mailbox_id)
+        self.clients[client_conn].remove(request)
+
+        # Forward internal cancel messages
+        addr = RuntimeAddress(-1, mailbox_id, 0)
+        self._handle_cancel(addr)
+
+        # Acknowledge the cancel request
+        if not client_conn.closed:
+            client_conn.send((RuntimeMessage.CANCEL, None))
+
+    def _handle_cancel(self, addr: RuntimeAddress) -> None:
+        """Cancel a runtime task in the system."""
+        for mconn in self.managers:
+            self.outgoing.append((mconn, RuntimeMessage.CANCEL, addr))
+
+    def _handle_error(self, error_payload: tuple[int, str]) -> None:
+        """Forward an error to the appropriate client and disconnect it."""
```

Comment:
Yeah, but when the client opens the error message, it will automatically raise it. So we anticipate the client will crash and stop all its other running tasks.